cpp/src/rolling/grouped_rolling.cu

```diff
@@ -371,26 +373,27 @@ std::unique_ptr<column> time_range_window_ASC(column_view const& input,
 /// Each group in the input timestamp column must be sorted,
 /// with null values clustered at either the start or the end of each group.
 /// If there are no nulls for any given group, (nulls_begin, nulls_end) == (0,0).
-std::tuple<rmm::device_vector<size_type>, rmm::device_vector<size_type>>
+std::tuple<rmm::device_uvector<size_type>, rmm::device_uvector<size_type>>
 get_null_bounds_for_timestamp_column(column_view const& timestamp_column,
-                                     rmm::device_uvector<size_type> const& group_offsets)
+                                     cudf::device_span<size_type const> group_offsets,
+                                     rmm::cuda_stream_view stream)
 {
   // For each group, the null values are themselves clustered
   // at the beginning or the end of the group.
   // These nulls cannot participate, except in their own window.
 
   // If the input has n groups, group_offsets will have n+1 values.
   // null_start and null_end should eventually have 1 entry per group.
-  auto null_start = rmm::device_vector<size_type>(group_offsets.begin(), group_offsets.end() - 1);
```

Comment:
We don't need to copy `group_offsets` to the device when we have nulls in `timestamp_column` as the `for_each` will write to every element in `null_start` and `null_end`

changes:
```
   // If the input has n groups, group_offsets will have n+1 values.
   // null_start and null_end should eventually have 1 entry per group.
-  auto null_start = cudf::detail::make_device_uvector_async(group_offsets, stream);
-  auto null_end   = cudf::detail::make_device_uvector_async(group_offsets, stream);
-
   if (timestamp_column.has_nulls()) {
+    auto null_start = cudf::detail::make_device_uvector_async(group_offsets.size(), stream);
+    auto null_end   = cudf::detail::make_device_uvector_async(group_offsets.size(), stream);
+
     auto p_timestamps_device_view = column_device_view::create(timestamp_column, stream);
     auto num_groups               = group_offsets.size();
 
@@ -430,9 +430,12 @@ get_null_bounds_for_timestamp_column(column_view const& timestamp_column,
             [&d_timestamps] __device__(auto i) { return d_timestamps.is_valid_nocheck(i); });
         }
       });
+
+    return std::make_tuple(std::move(null_start), std::move(null_end));
   }
 
-  return std::make_tuple(std::move(null_start), std::move(null_end));
+  return std::make_tuple(cudf::detail::make_device_uvector_async(group_offsets, stream),
+                         cudf::detail::make_device_uvector_async(group_offsets, stream));
 }
 
 // Time-range window computation, for timestamps in ASCENDING order.
```

Reply:
> `auto null_start = cudf::detail::make_device_uvector_async(group_offsets.size(), stream);`

That function doesn't exist, but I get the point. (I can just use the `device_uvector` ctor).

Reply:
Done.